homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,116 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    async_add_entities(ZimiLight(device, api) for device in api.lights)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_name = light.name.strip()
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        if self._light.type == "dimmer":
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        else:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._attr_name,
```

Comment:
```suggestion
            name=light.name.strip(),
```
so the entity name follows device name

Reply:
See comment on changes for line 50

Reply:
Did you try it? If the Home Assistant device has the correct humanly readable name, if you set the entity's name to `None` the name of the light entity should be the same as the Home Assistant device

Reply:
I did try.   The behaviour is as follows:

Firstly teh devices are created at HA startup and intially displayed in the UI with a displayed that is derived from the entity name.   The actual name is blank.  See next two screenshots.

![image](https://github.com/user-attachments/assets/fdae01fb-250e-438b-949f-fcfa36e16479)

![image](https://github.com/user-attachments/assets/318e4e9c-7acb-4427-8dbf-c3f93b1f125c)

But, after I operate the switch the name is updated in the UI only.

![image](https://github.com/user-attachments/assets/78b9f83e-fb5f-4d6a-8fca-2e71f3663959)

![image](https://github.com/user-attachments/assets/2bb33e8b-bd5b-44fa-81d0-6a9a1ae99068)


(For comparison, see the same device in my HACS integration version below where the human entered name, in this case 'Light' is displayed in the UI and is in the entity)

![image](https://github.com/user-attachments/assets/770d9279-696d-42de-b0a6-3f71ebd45dab)

![image](https://github.com/user-attachments/assets/4630ceb9-be25-47f7-bcc5-9c6a9581fad9)

Reply:
You are not showing the device, only the entity. Maybe something is wrong with `light.name`. The name of the switch being empty in the UI is fine. 
Here one example how it looks like:
![IMG_5192](https://github.com/user-attachments/assets/0bfde028-5834-41ee-9a24-ba5ec04fc110)
At the top you have the device name and on the bottom the lock which has name none and follows the device name. It's name in the UI is also empty

Reply:
Finally fixed.   See https://github.com/home-assistant/core/pull/129876/commits/97830b811057e09510a8782ba70fd9be96ee9569

The missing piece was that I needed self._attr_has_entity_name = True

Now the is showing the correct name that is pulled from the device rather than entity